packages/manager/src/features/CloudPulse/Utils/UserPreference.ts

```diff
@@ -1,86 +1,81 @@
-  if (!userPreference) {
-    userPreference = {} as AclpConfig;
-  }
-  const keys = Object.keys(data);
+interface AclpPreferenceObject {
+  isLoading: boolean;
+  preferences: AclpConfig;
+  updateGlobalFilterPreference: (data: AclpConfig) => void;
+  updateWidgetPreference: (label: string, data: Partial<AclpWidget>) => void;
+}
 
-  if (keys.includes(DASHBOARD_ID)) {
-    userPreference = { ...data, [TIME_DURATION]: userPreference.timeDuration };
-  } else {
-    userPreference = { ...userPreference, ...data };
-  }
+export const useAclpPreference = (): AclpPreferenceObject => {
+  const { data: preferences, isLoading } = usePreferences();
 
-  debounce(userPreference);
-};
+  const { mutateAsync: updateFunction } = useMutatePreferences();
 
-export const updateWidgetPreference = (
-  label: string,
-  data: Partial<AclpWidget>
-) => {
-  if (!userPreference) {
-    userPreference = {} as AclpConfig;
-  }
+  const preferenceRef = useRef<AclpConfig>(preferences?.aclpPreference ?? {});
 
-  if (!userPreference.widgets) {
-    userPreference.widgets = {};
+  if (preferences?.aclpPreference) {
+    preferenceRef.current = preferences.aclpPreference;
   }
-
-  userPreference.widgets[label] = {
-    ...userPreference.widgets[label],
-    label,
-    ...data,
+  /**
+   *
+   * @param data AclpConfig data to be updated in preferences
+   */
+  const updateGlobalFilterPreference = (data: AclpConfig) => {
+    let currentPreferences = { ...preferenceRef.current };
+    const keys = Object.keys(data);
+
+    if (keys.includes(DASHBOARD_ID)) {
+      currentPreferences = {
+        ...data,
+        [TIME_DURATION]: currentPreferences[TIME_DURATION],
+        [WIDGETS]: {},
+      };
+    } else {
+      currentPreferences = {
+        ...currentPreferences,
+        ...data,
+      };
+    }
+    preferenceRef.current = currentPreferences;
+    updateFunction({ aclpPreference: currentPreferences });
   };
 
-  debounce(userPreference);
-};
-
-// to avoid frequent preference update calls within 500 ms interval
-const debounce = (updatedData: AclpConfig) => {
-  if (timerId) {
-    clearTimeout(timerId);
-  }
-
-  timerId = setTimeout(() => useUpdateUserPreference(updatedData), 500);
+  /**
+   *
+   * @param label label of the widget that should be updated
+   * @param data AclpWidget data for the label that is to be updated in preference
+   */
+  const updateWidgetPreference = (label: string, data: Partial<AclpWidget>) => {
+    // sync with latest preferences
+    const updatedPreferences = {
+      ...preferenceRef.current,
+      [WIDGETS]: {
+        ...(preferenceRef.current.widgets ?? {}),
+      },
+    };
+    updatedPreferences.widgets[label] = {
+      ...updatedPreferences.widgets[label],
+      label,
+      ...data,
+    };
+
+    preferenceRef.current = updatedPreferences;
+    updateFunction({ aclpPreference: updatedPreferences });
+  };
+  return {
+    isLoading,
+    preferences: { ...(preferences?.aclpPreference ?? {}) },
```

Comment:
What is going on here? Can this be

```suggestion
    preferences: preferences?.aclpPreference ?? {},
```